File: CleverTapSDK/CleverTapEventDetail.m
#import "CleverTapEventDetail.h"

@implementation CleverTapEventDetail

- (NSString*) description {
    return [NSString stringWithFormat:@"CleverTapEventDetail (event name = %@; normalized event name = %@; first time = %d, last time = %d; count = %lu; device ID = %@)",
            self.eventName, self.normalizedEventName, (int) self.firstTime, (int) self.lastTime, (unsigned long)self.count, self.deviceID];
}

@end

File: CleverTapSDK/CleverTapEventDetail.swift
//
//  CleverTapEventDetail.swift
//  CleverTapSDK
//

import Foundation

@objc(CleverTapEventDetail)
@objcMembers
public class CleverTapEventDetail: NSObject {
    public var eventName: String?
    public var normalizedEventName: String?
    public var firstTime: TimeInterval = 0
    public var lastTime: TimeInterval = 0
    public var count: UInt = 0
    public var deviceID: String?

    public override var description: String {
        return "CleverTapEventDetail (event name = \(eventName ?? "nil"); normalized event name = \(normalizedEventName ?? "nil"); first time = \(Int(firstTime)), last time = \(Int(lastTime)); count = \(count); device ID = \(deviceID ?? "nil"))"
    }
}
